synapse/http/client.py

```diff
@@ -380,13 +380,15 @@ def post_urlencoded_get_raw(self, url, args={}):
 class SpiderEndpointFactory(object):
     def __init__(self, hs):
         self.blacklist = hs.config.url_preview_ip_range_blacklist
+        if hasattr(hs.config, "url_preview_ip_range_whitelist"):
+            self.whitelist = hs.config.url_preview_ip_range_whitelist
```

Comment:
You will want to set ``self.whitelist`` even if there isn't a ``url_preview_ip_range_whitelist`` in the config.

Perhaps ``self.whitelist = getattr(hs.config, "url_preview_ip_range_whitelist", None)``?